src/wowman/core.clj

```diff
@@ -579,6 +594,26 @@
 ;;
 
 
+(defn-spec get-create-user-catalog ::sp/catalog
+  []
+  (let [user-catalog-path (paths :user-catalog-file)]
+    (catalog/read-catalog
+     (if (fs/exists? user-catalog-path)
+       user-catalog-path
+       (catalog/write-empty-catalog! user-catalog-path)))))
```

Comment:
doesn't `write-empty-catalog!` return a path and not a catalog?

Reply:
it's wrapped in `read-catalog` which itself is a wrapper for loading and returning json. all good